### PR TITLE
Pubstwo 1310

### PR DESCRIPTION
**Pull request by mbucknell**
Introducing the ability to run the jasmine tests using the karma test runner, rather than using the maven plugin. I have also added the ability to use Travis CI to run the tests (python and jasmine). Developers should not merge code that has failed the Travis CI build. A failed build will produce a status message on the pull request page.

I have updated the README.md and dev_install.sh scripts with the additional steps needed to install the test runner and to run the karma tests.

### CODE REVIEW

**Comment by mbucknell** [comment]
I have not yet removed the maven-jasmine-plugin from the pom or its configuration file. I'm inclined to remove these things. The karma test runner is not as nice in terms of a nice display when running tests in the browser, but it is still possible to do the debugging via the karma DEBUG window and there are ways to run a single spec or describe by using fdescribe (or fit) on the jasmine spec or tests that you want to run. (see http://jasmine.github.io/2.3/focused_specs.html)

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['jasmine', 'requirejs'],
+
+
+    // list of files / patterns to load in the browser
+    files: [
+        'test-main.js',
+      {pattern: 'pubs_ui/bower_components/Squire.js/src/Squire.js', included: false},
```
Comment (mbucknell) [inline karma.conf.js]: I found it easiest to specify the specific file for the javascript dependencies. This means you have to add them here and in test-main.js. If you try to use wildcards, you end up having to exclude files and not all of the dependancies follow the same directory structure.

File: pubs_ui/lettuce_testing/features/test_utils.feature [hunk 2]
```diff
@@ -38,11 +38,6 @@ Feature: Test all pub_ui utilities
         Given I have a index page links that point to USGS and NGMDB and has plates
         When I manipulate the links with create display links
         Then I get the rejiggered display links that I expect
-
```
Comment (mbucknell) [inline pubs_ui/lettuce_testing/features/test_utils.feature]: Can't run this test from Travis CI since it hits a local URL.

**Comment by mbucknell** [comment]
I have assigned this to Mark but @instantsoup, @ayan-usgs , @dsteinich , if you have time please take a look. I would like some feedback on taking out the jasmine-maven-plugin in particular.

**Comment by mamcderm-usgs** [comment]
I don't have enough background/experience to provide any feedback on taking out the jasmine-maven-plugin, but what's here look good to me :+1:

**Pull request merged**